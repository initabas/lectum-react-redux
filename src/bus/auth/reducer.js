// Instruments
import types from './types';
import { Map } from 'immutable';

const initialState = Map({
    authenticated: false,
});

export default (state = initialState, action) => {
    switch (action.type) {
        case types.AUTHENTICATE_SUCCESS:
        case types.LOGIN_SUCCESS:
        case types.SIGNUP_SUCCESS:
            return state.set('authenticated', true);
        case types.LOGOUT_SUCCESS:
            return state.set('authenticated', false);
        default:
            return state;
    }
};
