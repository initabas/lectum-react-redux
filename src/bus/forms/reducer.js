import { combineForms } from 'react-redux-form';

export default combineForms({
    login: {
        email:    'sasha.melnik@gmail.com',
        password: '',
        remember: false,
    },
    signup: {
        firstName: 'sasha',
        lastName:  'melnyk',
        email:     'sasha.melnik@gmail.com',
        password:  '11111111',
        invite:    'f50h1UtrtdgS',
    },
    user: {
        profile: {
            avatar:    '',
            firstName: '',
            lastName:  '',
        },
        password: {
            oldPassword: '',
            newPassword: '',
        }
    },
}, 'forms');
