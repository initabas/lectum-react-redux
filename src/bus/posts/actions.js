import types from './types';

export const postsActions = Object.freeze({
    fetchPosts: () => ({
        type:    types.FETCH_POSTS,
    }),
    fetchPostsSuccess: (posts) => ({
        type:    types.FETCH_POSTS_SUCCESS,
        payload: posts,
    }), 
    fetchPostsFail: (error) => ({
        type:    types.FETCH_POSTS_FAIL,
        payload: error,
        error:   true,
    }),
    
    createPost: (comment) => ({
        type:    types.CREATE_POST,
        payload: comment,
    }),
    createPostSuccess: (post) => ({
        type:    types.CREATE_POST_SUCCESS,
        payload: post,
    }), 
    createPostFail: (error) => ({
        type:    types.CREATE_POST_FAIL,
        payload: error,
        error:   true,
    }),

    deletePost: (id) => ({
        type:    types.DELETE_POST,
        payload: id,
    }),
    deletePostSuccess: (id) => ({
        type:    types.DELETE_POST_SUCCESS,
        payload: id,
    }), 
    deletePostFail: (error) => ({
        type:    types.DELETE_POST_FAIL,
        payload: error,
        error:   true,
    }),

    likePost: (postId) => ({
        type:    types.LIKE_POST,
        payload: postId,
    }),
    likePostSuccess: (likedPostIds) => ({
        type:    types.LIKE_POST_SUCCESS,
        payload: likedPostIds,
    }), 
    likePostFail: (error) => ({
        type:    types.LIKE_POST_FAIL,
        payload: error,
        error:   true,
    }),

    dislikePost: (postId) => ({
        type:    types.DISLIKE_POST,
        payload: postId,
    }),
    dislikePostSuccess: (dislikedPostIds) => ({
        type:    types.DISLIKE_POST_SUCCESS,
        payload: dislikedPostIds,
    }), 
    dislikePostFail: (error) => ({
        type:    types.DISLIKE_POST_FAIL,
        payload: error,
        error:   true,
    }),
});
