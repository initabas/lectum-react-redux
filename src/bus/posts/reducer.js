// Instruments
import types from './types';
import { List, fromJS } from 'immutable';

const initialState = List([]);

export default (state = initialState, action) => {
    switch (action.type) {
        case types.FETCH_POSTS_SUCCESS:
            return fromJS(action.payload);
            
        case types.CREATE_POST_SUCCESS:
            return state.unshift(fromJS(action.payload));
            
        case types.DELETE_POST_SUCCESS:
            return state.filter((post) => post.get('id') !== action.payload);

        case types.LIKE_POST_SUCCESS:
            const postIdToLike = state.findIndex((post) => post.get('id') === action.payload.postId);
            return state.updateIn(
                [postIdToLike, 'likes'], (likedBy) => 
                    likedBy.push(action.payload.user)
            );

        case types.DISLIKE_POST_SUCCESS:
            const postIdToDislike = state.findIndex((post) => post.get('id') === action.payload.postId);
            return state.updateIn(
                [postIdToDislike, 'likes'], (likedBy) => 
                    likedBy.filter((me) => me.get('id') !== action.payload.userId)
            );

        default:
            return state;
    }
};
