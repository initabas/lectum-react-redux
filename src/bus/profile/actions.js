import types from './types';

export const profileActions = Object.freeze({
    fillProfile: (profile) => ({
        type:    types.FILL_PROFILE,
        payload: profile,
    }),
});
