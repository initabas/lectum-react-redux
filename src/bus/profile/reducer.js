import { Map } from 'immutable';

// Instruments
import types from './types';

const initialState = Map({
    id:        '',
    lastName:  '',
    firstName: '',
    avatar:    '',
});

export default (state = initialState, action) => {
    switch (action.type) {
        case types.FILL_PROFILE:
            return state.merge(action.payload);

        default:
            return state;
    }
};
