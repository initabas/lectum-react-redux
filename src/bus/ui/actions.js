import types from './types';

export const uiActions = Object.freeze({
    setFeedFetchingState: (state) => ({
        type:    types.SET_FEED_FETCHING_STATE,
        payload: state,
    }),
    initialize: (state) => ({
        type:    types.INITIALIZE,
    }),
    setOnLineStatus: (state) => ({
        type:    types.SET_ONLINE_STATUS,
        payload: state,
    }),
    setAuthFetchingState: (state) => ({
        type:    types.SET_AUTH_FETCHING_STATE,
        payload: state,
    }),
});
