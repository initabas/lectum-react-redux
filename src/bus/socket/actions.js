import { socket } from 'init/socket';
import { uiActions } from '../ui/actions';

export const watchConnection = () => (dispatch) => {
    socket.on('connect', () => {
        dispatch(uiActions.setOnLineStatus(true));
    });
    socket.on('disconect', () => {
        dispatch(uiActions.setOnLineStatus(false));
    });
};

export const watchPosts = () => () => {
    // socket.on('connect', () => {
    //     dispatch(uiActions.setOnLineStatus(true));
    // });
    // socket.on('disconect', () => {
    //     dispatch(uiActions.setOnLineStatus(false));
    // });
};
