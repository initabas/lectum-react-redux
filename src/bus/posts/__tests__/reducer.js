import { postsActions } from '../actions';
import { fromJS } from 'immutable';

const state = fromJS(__.posts);
const user = fromJS(__.profile);

import reducer from '../reducer';

describe('posts reducers:', () => {
    test('should return initial state', () => {
        // void 0 === undefined
        expect(reducer(void 0, {})).toMatchSnapshot();
    });

    test('should handle FETCH_POSTS_SUCCESS', () => {
        expect(reducer(void 0, postsActions.fetchPostsSuccess(__.posts))
        ).toMatchSnapshot();
    });

    test('should handle CREATE_POST_SUCCESS', () => {
        expect(reducer(void 0, postsActions.createPostSuccess(__.post1))
        ).toMatchSnapshot();
    });

    test('should handle DELETE_POST_SUCCESS', () => {
        expect(reducer(state, postsActions.deletePostSuccess(__.post2))
        ).toMatchSnapshot();
    });

    test('should handle LIKE_POST_SUCCESS', () => {
        expect(reducer(state, postsActions.likePostSuccess({
            postId: __.post1.id,
            user,
        }))
        ).toMatchSnapshot();
    });
    
    test('should handle DISLIKE_POST_SUCCESS', () => {
        expect(reducer(state, postsActions.dislikePostSuccess({
            postId: __.post1.id,
            userId: __.post1.likes[0].id,
        }))
        ).toMatchSnapshot();
    });
});
