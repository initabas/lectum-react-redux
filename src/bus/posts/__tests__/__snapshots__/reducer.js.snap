// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts reducers: should handle CREATE_POST_SUCCESS 1`] = `
Immutable.List [
  Immutable.Map {
    "id": "59edd93052db76aefs2b36ee_TEST",
    "comment": "Iusto aut quae non asperiores eius illo ut.",
    "created": 1508760000,
    "likes": Immutable.List [
      Immutable.Map {
        "id": "52edd9352db76aeff2b34ed",
        "firstName": "Harold",
        "lastName": "Cassin",
      },
    ],
    "author": Immutable.Map {
      "id": "34edd9c052db76aeff2b33es",
      "firstName": "Tristian",
      "lastName": "Farrell",
      "groupId": "844t1zdqp0",
      "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
    },
  },
]
`;

exports[`posts reducers: should handle DELETE_POST_SUCCESS 1`] = `
Immutable.List [
  Immutable.Map {
    "id": "59edd93052db76aefs2b36ee_TEST",
    "comment": "Iusto aut quae non asperiores eius illo ut.",
    "created": 1508760000,
    "likes": Immutable.List [
      Immutable.Map {
        "id": "52edd9352db76aeff2b34ed",
        "firstName": "Harold",
        "lastName": "Cassin",
      },
    ],
    "author": Immutable.Map {
      "id": "34edd9c052db76aeff2b33es",
      "firstName": "Tristian",
      "lastName": "Farrell",
      "groupId": "844t1zdqp0",
      "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
    },
  },
  Immutable.Map {
    "id": "59edd93052db76aefs2b36ed_TEST",
    "comment": "Iusto aut quae non asperiores eius illo ut.",
    "created": 1508760000,
    "likes": Immutable.List [
      Immutable.Map {
        "id": "52edd9352db76aeff2b34ed",
        "firstName": "Harold",
        "lastName": "Cassin",
      },
    ],
    "author": Immutable.Map {
      "id": "34edd9c052db76aeff2b33es",
      "firstName": "Tristian",
      "lastName": "Farrell",
      "groupId": "844t1zdqp0",
      "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
    },
  },
]
`;

exports[`posts reducers: should handle DISLIKE_POST_SUCCESS 1`] = `
Immutable.List [
  Immutable.Map {
    "id": "59edd93052db76aefs2b36ee_TEST",
    "comment": "Iusto aut quae non asperiores eius illo ut.",
    "created": 1508760000,
    "likes": Immutable.List [],
    "author": Immutable.Map {
      "id": "34edd9c052db76aeff2b33es",
      "firstName": "Tristian",
      "lastName": "Farrell",
      "groupId": "844t1zdqp0",
      "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
    },
  },
  Immutable.Map {
    "id": "59edd93052db76aefs2b36ed_TEST",
    "comment": "Iusto aut quae non asperiores eius illo ut.",
    "created": 1508760000,
    "likes": Immutable.List [
      Immutable.Map {
        "id": "52edd9352db76aeff2b34ed",
        "firstName": "Harold",
        "lastName": "Cassin",
      },
    ],
    "author": Immutable.Map {
      "id": "34edd9c052db76aeff2b33es",
      "firstName": "Tristian",
      "lastName": "Farrell",
      "groupId": "844t1zdqp0",
      "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
    },
  },
]
`;

exports[`posts reducers: should handle FETCH_POSTS_SUCCESS 1`] = `
Immutable.List [
  Immutable.Map {
    "id": "59edd93052db76aefs2b36ee_TEST",
    "comment": "Iusto aut quae non asperiores eius illo ut.",
    "created": 1508760000,
    "likes": Immutable.List [
      Immutable.Map {
        "id": "52edd9352db76aeff2b34ed",
        "firstName": "Harold",
        "lastName": "Cassin",
      },
    ],
    "author": Immutable.Map {
      "id": "34edd9c052db76aeff2b33es",
      "firstName": "Tristian",
      "lastName": "Farrell",
      "groupId": "844t1zdqp0",
      "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
    },
  },
  Immutable.Map {
    "id": "59edd93052db76aefs2b36ed_TEST",
    "comment": "Iusto aut quae non asperiores eius illo ut.",
    "created": 1508760000,
    "likes": Immutable.List [
      Immutable.Map {
        "id": "52edd9352db76aeff2b34ed",
        "firstName": "Harold",
        "lastName": "Cassin",
      },
    ],
    "author": Immutable.Map {
      "id": "34edd9c052db76aeff2b33es",
      "firstName": "Tristian",
      "lastName": "Farrell",
      "groupId": "844t1zdqp0",
      "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
    },
  },
]
`;

exports[`posts reducers: should handle LIKE_POST_SUCCESS 1`] = `
Immutable.List [
  Immutable.Map {
    "id": "59edd93052db76aefs2b36ee_TEST",
    "comment": "Iusto aut quae non asperiores eius illo ut.",
    "created": 1508760000,
    "likes": Immutable.List [
      Immutable.Map {
        "id": "52edd9352db76aeff2b34ed",
        "firstName": "Harold",
        "lastName": "Cassin",
      },
      Immutable.Map {
        "firstName": "Walter",
        "lastName": "White",
        "id": "123",
        "avatar": "avatar",
        "token": "NOT_A_TOKEN",
      },
    ],
    "author": Immutable.Map {
      "id": "34edd9c052db76aeff2b33es",
      "firstName": "Tristian",
      "lastName": "Farrell",
      "groupId": "844t1zdqp0",
      "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
    },
  },
  Immutable.Map {
    "id": "59edd93052db76aefs2b36ed_TEST",
    "comment": "Iusto aut quae non asperiores eius illo ut.",
    "created": 1508760000,
    "likes": Immutable.List [
      Immutable.Map {
        "id": "52edd9352db76aeff2b34ed",
        "firstName": "Harold",
        "lastName": "Cassin",
      },
    ],
    "author": Immutable.Map {
      "id": "34edd9c052db76aeff2b33es",
      "firstName": "Tristian",
      "lastName": "Farrell",
      "groupId": "844t1zdqp0",
      "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
    },
  },
]
`;

exports[`posts reducers: should return initial state 1`] = `Immutable.List []`;
